src/ServiceControl.Monitoring/Retries/RetriesStore.cs

```diff
@@ -1,7 +1,28 @@
 ﻿namespace ServiceControl.Monitoring
 {
+    using System;
     using Infrastructure;
-    public class RetriesStore : VariableHistoryIntervalStore
+    using Messaging;
+
+    public class RetriesStore : IProvideBreakdownBy<EndpointInstanceId>, IProvideBreakdownBy<EndpointMessageType>
     {
+        VariableHistoryIntervalStore<EndpointInstanceId> byInstance { get; } = new VariableHistoryIntervalStore<EndpointInstanceId>();
+        VariableHistoryIntervalStore<EndpointMessageType> byMessageType { get; } = new VariableHistoryIntervalStore<EndpointMessageType>();
+
+        public IntervalsStore<EndpointInstanceId>.IntervalsBreakdown[] GetIntervals(HistoryPeriod period, DateTime now)
+        {
+            return byInstance.GetIntervals(period, now);
```

Comment:
Given that this implementation is the same for every implementation of `IProvideBreakDownBy<>` we could consider just exposing the property on the interface. Each _store_ would need to explicitly implement each of the interfaces but that becomes one-line. For every downstream that needs to consume it you would do

```
stores.OfType<TBreakdown>().Select(x => x.Breakdown.GetIntervals(period, now));
```

Just a thought.

Reply:
It does get messier to access them internally though